### PR TITLE
Feature: APP-1073 - Proposal onTransaction Detail

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description
- adds proposal title on `transactionDetail` modal
- fixes app hang up when invalid proposal id is used on Proposal Details page
- refactors `Transfer` type for more granularity

Note: ~Because SDK is returning an invalid proposal id for the withdraw transfer, when reviewing please replace the proposal id [here](https://github.com/aragon/apps/blob/f37bc768de3bb7b08a1d9865c6cf0c1aea7c923b/packages/web-app/src/containers/transactionDetail/index.tsx#L47) with `0x3e16a08ad61500b2bc77ddcc04387f0fabc04469_0x0`~
Okay, so I lied; fight me.

Task: [APP-1073](https://aragonassociation.atlassian.net/browse/APP-1073)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/containers/transactionDetail/index.tsx [hunk 1]
```diff
@@ -64,13 +94,23 @@ const TransactionDetail: React.FC<TransactionDetailProps> = ({daoName}) => {
...
+          ? transfer.reference && (
+              <CardText
+                type="label"
+                title={t('labels.reference')}
+                content={transfer.reference}
+              />
+            )
+          : transfer.transferType === TransferTypes.Withdraw && (
+              <ListItemAction
+                title={proposal?.metadata.title || 'Loading...'}
```
Comment (sepehr2github) [inline packages/web-app/src/containers/transactionDetail/index.tsx]: Could you use a translated string here, Thanks

**Review by sepehr2github — APPROVED** [review]
Thanks, Fabrice! Left a minor comment, Seems good to go

File: packages/web-app/src/containers/transactionDetail/index.tsx [hunk 2]
```diff
@@ -64,13 +94,23 @@ const TransactionDetail: React.FC<TransactionDetailProps> = ({daoName}) => {
...
-        {transfer.reference && (
-          <CardText
-            type="label"
-            title={t('labels.reference')}
-            content={transfer.reference}
-          />
-        )}
+
+        {transfer.transferType === TransferTypes.Deposit
+          ? transfer.reference && (
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/transactionDetail/index.tsx:99]: Why not make the reference mandatory, then? Or is it a deposit vs. withdraw attribute? Could we just move it to withdraw then?

**Review by WissenIstNacht — APPROVED** [review]
Hey there, looks good!

File: packages/web-app/src/containers/transactionDetail/index.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/transactionDetail/index.tsx:99]: Reference isn't a mandatory input when creating a deposit/withdraw

**Pull request merged**